### PR TITLE
python: Clean up children of spawned processes

### PR DESCRIPTION
**Pull request by martinpitt**
Set `prctl(SET_PDEATHSIG, SIGHUP)` for cockpit.spawn() and friends, to clean up subprocesses of the spawned process. The C bridge does that also (in cockpitpipe.c).

### CODE REVIEW

**Comment by martinpitt** [comment]
[some flakes](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18185-20230117-081855-bd66b0be-fedora-37-pybridge/log.html#198) and two fixed tests. Comparing..

File: src/cockpit/channels/stream.py [hunk 1]
```diff
@@ -84,7 +96,8 @@ def create_transport(self, loop: asyncio.AbstractEventLoop, options: Dict[str, A
 
         try:
             logger.debug('Spawning process args=%s', args)
-            return SubprocessTransport(loop, self, args, pty, window, env=env, cwd=cwd, stderr=stderr)
+            return SubprocessTransport(loop, self, args, pty, window, env=env, cwd=cwd, stderr=stderr,
+                                       preexec_fn=lambda: prctl(SET_PDEATHSIG, signal.SIGHUP))
```
Comment (allisonkarlitskaya) [inline src/cockpit/channels/stream.py:100]: Why not `SIGKILL`?

**Review by allisonkarlitskaya — APPROVED** [review]
Looks totally fine to me, but one question.  Is there any reason we might want graceful shutdown on our subprocesses?  HUP seems like a strange choice, in particular...

File: src/cockpit/channels/stream.py (hunk 1 above)
Comment (martinpitt) [reply src/cockpit/channels/stream.py:100]: Good question -- hysterical raisins mostly, the C bridge [uses SIGHUP](https://github.com/cockpit-project/cockpit/blob/main/src/common/cockpitpipe.c#L1272). It's been like that forever, see commit aa407c961d98e.

SIGKILL seems a little aggressive, SIGTERM may be a good alternative.

So.. I dunno

**Review by allisonkarlitskaya — APPROVED** [review]
Note: this conflicts #18183.  I marked it blocked on this.  Please land here first, and then @jelly can rebase.

**Comment by martinpitt** [comment]
The [two failures](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18185-20230117-164632-269e4108-fedora-37-pybridge/log.html#315) due to unexpected messages are due to [this regression on main](https://github.com/cockpit-project/cockpit/pull/18188#issuecomment-1385703320), unrelated.

**Pull request merged**